lib/board.rb

```diff
@@ -0,0 +1,87 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    # @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    @cells = Array.new(9, " ")
+  end
+
+  def position(position_number)
+    case position_number
+    when "1"
+      self.cells[0]
+    when "2"
+      self.cells[1]
+    when "3"
+      self.cells[2]
+    when "4"
+      self.cells[3]
+    when "5"
+      self.cells[4]
+    when "6"
+      self.cells[5]
+    when "7"
+      self.cells[6]
+    when "8"
+      self.cells[7]
+    when "9"
+      self.cells[8]
+    end
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{self.position("1")} | #{self.position("2")} | #{self.position("3")} "
+    puts "-----------"
+    puts " #{self.position("4")} | #{self.position("5")} | #{self.position("6")} "
+    puts "-----------"
+    puts " #{self.position("7")} | #{self.position("8")} | #{self.position("9")} "
+  end
+
+  def full?
+    self.cells.all? {|character| character == "X" || character == "O"}
+  end
+
+  def turn_count
+    counter = 0
+    self.cells.each do |position|
+      if position == "X" || position == "O"
+      counter += 1
+      end
+    end
+    counter
+  end
+
+  def taken?(position_number)
```

Comment:
is position number always equal to the number the user inputs or do you at some point change it to be `-1`? just noticed that sometimes you use position number - 1 and sometimes you don't? This is confusing.